libbeat/processors/add_host_metadata/add_host_metadata.go

```diff
@@ -81,6 +81,25 @@ func New(cfg *common.Config) (processors.Processor, error) {
 
 // Run enriches the given event with the host meta data
 func (p *addHostMetadata) Run(event *beat.Event) (*beat.Event, error) {
+	// check replace_host_fields field
+	if !p.config.ReplaceHostFields {
+		// If host fields exist(besides host.name added by libbeat) in event, skip add_host_metadata.
+		hostFields, err := event.Fields.GetValue("host")
+		if err == nil && hostFields != nil && len(hostFields.(common.MapStr)) >= 1 {
```

Comment:
nit: maybe consider doing the cast only once (`hostFields = hostFields.(common.MapStr)`)